### ISSUE TITLE
During inheritance, state accessors don't conflict with parent functions

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#1084]
```
contract Interface {
    function isObsolete() returns (bool);
}

contract Test is Interface {
    bool public isObsolete;
}
```

The conflict is not detected, and both of these contracts are considered abstract by the compiler (no bytecode is returned).

(Also: in the interface, the `public` modifier will implicitly set the `constant` modifier to true. I think we should document this.)

---

When including both in a single contract, the conflict is properly detected:

```
contract Test {
    bool public isObsolete;
    function isObsolete() returns (bool);
}
```

```
Untitled6:3:5: Error: Identifier already declared.
    function isObsolete() returns (bool);
    ^-----------------------------------^
The previous declaration is here: Untitled6:2:5: 
    bool public isObsolete;
    ^--------------------^
```

---

If there `Interface` has a body, the child will (properly?) overrides the `isObsolete` method.

```
contract Interface {
    function isObsolete() returns (bool) { return false; }
}

contract Test is Interface {
    bool public isObsolete = true;
}
```

**Comment by pirapira** [comment]
#1292 fixes this issue.

**Comment by pirapira** [comment]
In the first example, I think the compiler is doing the right thing because the derived function is still abstract.  `isObsolete` function is only defined externally, but not for internal calls.
```
contract Interface {
    function isObsolete() returns (bool);
}

contract Test is Interface {
    bool public isObsolete;
    function user() {
        isObsolete();  // not yet implemented.
    }
}
```

**Comment by pirapira** [comment]
I'm not totally sure what behavior I should expect from the second example.

```
contract Interface {
    function isObsolete() returns (bool) { return false; }
}

contract Test is Interface {
    bool public isObsolete = true;
    function internal() {
        isObsolete();  // should this still return false?
    }
}
```

**Comment by pirapira** [comment]
@chriseth I need an opinion about my last comment above, about `isObsolete()` function.

**Comment by chriseth** [comment]
Even if it is confusing, I think we have to make a distinction between the internal inheritance structure and the external one. Because of that, I think the current way the compiler works is fine.

**Comment by chriseth** [comment]
This has been fixed in 0.4.9.
